app/services/discover_integration_service.rb

```diff
@@ -18,14 +18,16 @@ def call(integration)
     klass = case integration
             when Integration::Keycloak
               Integration::KeycloakService
+            when Integration::Generic
+              Integration::GenericService
             when integration
               Integration::EchoService
             else # the only one for now
               raise NotImplementedError
-            end
+            end.freeze
```

Comment:
Is it expected to freeze classes at runtime? Can you explain me why? I do simply not get the use case

Reply:
I wanted to freeze the instance. Well spotted.